### DISCUSSION TITLE
Fetch tasks for given workflow name and project name

### DISCUSSION THREAD

**Discussion by najoshi7** [PrefectHQ/prefect#3606] [Q&A] [answered]
Hello prefect community,

I have a question:
How to fine tasks for a given workflow name (e.g. 'workflow-2') and project name (e.g. 'user1')? I tried following query in my python program:
query = {
    "query": {
        with_args(
            "task",
            {
                "where": {"flow": {"name": {"_eq": "workflow-2"}}}          ## i.e. return tasks belonging to a flow: workflow-2
            }
            ): {
                "name": True,
                "flow": {
                    "name": True,
                    "project('where': {'name' {'_eq': 'user1'}})": {              ## i.e. return tasks belonging to a project: user1
                        "name":True
                    }
                }
            }
        }
    }

The above query does not work say an error:
"requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:4200/graphql"

Thanks.

**Comment by joshmeek** [comment, marked as answer]
Hi @najoshi7 try something like this:

```python
query = {
    "query": {
        with_args(
            "flow",
            {
                "where": {
                    "_and": {
                        "project": {
                            "name": {"_eq": "user1"},
                        },
                        "name": {"_eq": "workflow-2"},
                    }
                }
            },
        ): {"name": True, "id": True}
    }
}

client.graphql(query)
```

**Reply by najoshi7** [reply]
Hi @joshmeek , I tried it. But it fetches workflows. 
However, I want to fetch task details. i.e. Display all those **tasks** which belong to the flow "workflow-2" and the project "user1".

Thanks.

**Reply by najoshi7** [reply]
Hi @joshmeek , I tried the following query also, but it results into an error bad request url: http://localhost:4200/graphql   :

```
query = {
    "query": {
        with_args(
            "task",
            {
                "where": {
                    "_and": {
                        "project": {
                            "name": {"_eq": "user1"},
                        },
                        "flow": {
                            "name": {"_eq": "workflow-2"}
                        }
                    }
                }
            },
        ): {"name": True, "id": True}
    }
}
```

Please suggest.

**Reply by joshmeek** [reply]
My bad, misread the original unformatted query. You're going to want something like this:

```python
query = {
    "query": {
        with_args(
            "task",
            {
                "where": {
                    "_and": {
                        "flow": {
                            "name": {"_eq": "workflow-2"},
                            "project": {
                                "name": {"_eq": "user1"}
                            }
                        },
                    }
                }
            },
        ): {"id": True}
    }
}
```

Project doesn't have a relationship to tasks but to flows so you need to grab the matching project name off of the flow in the where clause

**Reply by najoshi7** [reply]
@joshmeek 
Thanks joshmeek. It works. 
Thanks again for your help.
